pynumaflow/sink/server.py

```diff
@@ -0,0 +1,107 @@
+import asyncio
+import logging
+from os import environ
+
+
+from google.protobuf import empty_pb2 as _empty_pb2
+
+import grpc
+from typing import Callable, Any, Iterator, List
+
+from pynumaflow._constants import (
+    SINK_SOCK_PATH,
+    DATUM_KEY,
+)
+from pynumaflow.sink import Response, Responses, Datum
+from pynumaflow.sink.generated import udsink_pb2_grpc, udsink_pb2
+from pynumaflow.types import NumaflowServicerContext
+
+if environ.get("PYTHONDEBUG"):
+    logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+UDSinkCallable = Callable[[List[Datum], Any], Responses]
+
+
+class UserDefinedSinkServicer(udsink_pb2_grpc.UserDefinedSinkServicer):
+    """
+    Provides an interface to write a User Defined Sink (UDSink)
+    which will be exposed over gRPC.
+
+    Args:
+        sink_handler: Function callable following the type signature of UDSinkCallable
+        sock_path: Path to the UNIX Domain Socket
+
+    Example invocation:
+    >>> from typing import List
+    >>> from pynumaflow.sink import Datum, Responses, Response, UserDefinedSinkServicer
+    >>> def udsink_handler(datum_list: List[Datum]) -> Responses:
+    ...   responses = Responses()
+    ...   for msg in datum_list:
+    ...     print("User Defined Sink", msg)
+    ...     responses.append(Response.as_success(msg.id))
+    ...   return responses
+    >>> grpc_server = UserDefinedSinkServicer(udsink_handler)
+    >>> grpc_server.start()
+    """
+
+    def __init__(self, sink_handler: UDSinkCallable, sock_path=SINK_SOCK_PATH):
+        self.__sink_handler: UDSinkCallable = sink_handler
+        self.sock_path = sock_path
+        self._cleanup_coroutines = []
+
+    def SinkFn(
+        self, request: udsink_pb2.DatumList, context: NumaflowServicerContext
+    ) -> udsink_pb2.ResponseList:
+        """
+        Applies a sink function to a list of datum elements.
+        The camel case function name comes from the generated udsink_pb2_grpc.py file.
```

Comment:
This is Pascal case :) 
```suggestion
        The pascal case function name comes from the generated udsink_pb2_grpc.py file.
```